worldedit-core/src/main/java/com/sk89q/worldedit/extent/world/SurvivalModeExtent.java

```diff
@@ -91,8 +96,21 @@ public void setStripNbt(boolean stripNbt) {
     @Override
     public <B extends BlockStateHolder<B>> boolean setBlock(BlockVector3 location, B block) throws WorldEditException {
         if (toolUse && block.getBlockType().getMaterial().isAir()) {
-            world.simulateBlockMine(location);
-            return true;
+            Collection<BaseItemStack> drops = world.simulateBlockMine(location);
+            boolean canSet = super.setBlock(location, block);
+            if (canSet) {
+                for (BaseItemStack stack : drops) {
```

Comment:
Put the for loop in the `run` method instead.

Reply:
@SirYwell Would it be appropriate to return a boolean with ``#simulateBlockMine`` method, without changing its behavior?